### DISCUSSION TITLE
Using a custom html-validate rule

### DISCUSSION THREAD

**Discussion by Axel3005** [nuxt-modules/html-validator#45] [Q&A] [answered]
Hello, 

I would like to use html-validate custom rule (https://html-validate.org/dev/writing-rules.html), but i don't know how to call it on my plugins options. I'm using Nuxt 2.15.3.

My plugin options are : 

``` 
htmlValidator: {
    usePrettier: false,
    options: {
      extends: [
        'html-validate:recommended',
      ],
      rules: {
        "custom-rule": "error",
        "no-raw-characters": "warn",
        "unrecognized-char-ref": "warn",
        'attr-quotes': ['warn', {
          "style": "auto"
        }],
        'heading-level': 'off',
        'input-missing-label': 'off',
        'element-required-attributes': 'off',
        'element-permitted-content': "off"
      }
    }
  }
```

My custom rule is created on my project root, but on build i have an error 'Definition for rule was not found'. It seems looking for an html-validate rule and not my custom rule :
![Capture d’écran 2021-04-14 à 14 02 42](https://user-images.githubusercontent.com/82514430/114707109-1d423e80-9d2a-11eb-8f45-e853b908602b.png)

Didn't see anything about custom rule on the plugin documentation, if you have any clue it will be greatly appreciated

Thanks, 
Axel

**Comment by ext** [comment, marked as answer]
To add custom rules you must [create a plugin](https://html-validate.org/dev/writing-plugins.html) with the [rule definitions](https://html-validate.org/dev/writing-plugins.html#rules).

Something similar to

```js
/* CustomRule is a class inheriting from Rule, as per documentation */
const CustomRule = require("./rules/my-rule.js");

/* plugin which adds the "my-prefix/custom-rule" definition */
module.exports = {
  rules: {
    "my-prefix/custom-rule": AnotherRule,
  },
};
```

The plugin would be added like this:

```diff
       extends: [
         'html-validate:recommended',
       ],
+      plugins: [
+        "my-plugin" // npm package or local path
+      ],
       rules: {
-        "custom-rule": "error",
+        "my-prefix/custom-rule": "error",
         "no-raw-characters": "warn",
```
